src/util/heap/freelistpageresource.rs

```diff
@@ -262,52 +294,66 @@ impl<VM: VMBinding> FreeListPageResource<VM> {
     }
 
     fn allocate_contiguous_chunks(
-        &mut self,
+        &self,
```

Comment:
I am concerned about changes like this in this PR. The current code (before this PR) gets a mutable reference when we acquire a lock. In this case, we can reason that getting the mutable reference is correct, as we have the lock. In this PR, many methods are changed from `&mut self` to `&self`, and they get mutable references internally. If the caller is not aware of this, this could easily lead to a race condition.

Reply:
One way around this is mark these functions as unsafe, other way around is add mutating methods to `*Inner` types, what would be the best?

Reply:
Maybe just mark them as unsafe for now. We hope to do a safe implementation at some point.

Reply:
Marked these functions as unsafe now.